### PR TITLE
MGMT-2472 Format bootable disks on Install command

### PR DESCRIPTION
**Pull request by rollandf**

### CODE REVIEW

**Comment by rollandf** [comment]
/hold
Need to add unit-tests

**Comment by rollandf** [comment]
@filanov

**Comment by rollandf** [comment]
@eranco74 @ronniel1 @avishayt

File: internal/host/installcmd.go [hunk 1]
```diff
@@ -136,6 +139,21 @@ func (i *installCmd) GetSteps(ctx context.Context, host *models.Host) ([]*models
 	return []*models.Step{step}, nil
 }
 
+func getDiskFormatCmds(log logrus.FieldLogger, host models.Host) string {
+	var inventory models.Inventory
+	if err := json.Unmarshal([]byte(host.Inventory), &inventory); err != nil {
```
Comment (filanov) [inline internal/host/installcmd.go:144]: Maybe we should fail? i don't like that we hide issues.

**Review by filanov — COMMENTED** [review]
No tests?

**Comment by rollandf** [comment]
`Oct 21 13:36:45 test-infra-cluster-assisted-installer-master-1 agent[1762]: time="21-10-2020 13:36:45" level=info msg="Executing step: <install-9c6751a4>, command: <bash>, args: <[-c dd if=/dev/zero of=/dev/vdb bs=10M count=1 ; dd if=/dev/zero of=/dev/vdc bs=10M count=1 ; podman run -v /dev:/dev:rw -v /opt:/opt:rw -v /run/systemd/journal/socket:/run/systemd/journal/socket --privileged --pid=host --net=host -v /var/log:/var/log:rw --env PULL_SECRET_TOKEN --name assisted-installer quay.io/ocpmetal/assisted-installer:latest --role master --cluster-id c1ceb6fc-8535-45bf-b1bc-40f0290be373 --boot-device /dev/vda --host-id 13ef17fa-b4c8-4d15-835f-3274387eca0f --openshift-version 4.6 --controller-image quay.io/ocpmetal/assisted-installer-controller:latest --url http://10.1.37.42:6000 --insecure=true --agent-image quay.io/rollandf/bm-inventory:agentboot --host-name test-infra-cluster-assisted-installer-master-1]>" file="step_processor.go:71" request_id=f78f6df6-9476-427a-9b70-d3ba63ef457e
`

File: internal/host/installcmd.go (hunk 1 above)
Comment (rollandf) [reply internal/host/installcmd.go:148]: Fixed

**Comment by filanov** [comment]
LGTM @yevgeny-shnaidman @ronniel1 can you please review as well?

**Comment by yevgeny-shnaidman** [comment]
looks good

File: internal/host/installcmd.go [hunk 2]
```diff
@@ -136,6 +143,23 @@ func (i *installCmd) GetSteps(ctx context.Context, host *models.Host) ([]*models
...
+		log.Errorf("Failed to get inventory from host with id %s", host.ID)
+		return "", err
+	}
+	formatCmds := ""
+	for _, disk := range inventory.Disks {
+		if disk.Bootable {
+			dev := GetDeviceFullName(disk.Name)
+			formatCmds = formatCmds + fmt.Sprintf("dd if=/dev/zero of=%s bs=512 count=33 ;", dev)
+			formatCmds = formatCmds + fmt.Sprintf("dd if=/dev/zero of=%s bs=512 count=33 seek=$((`blockdev --getsz %s` - 33)); ", dev, dev)
```
Comment (avishayt) [inline internal/host/installcmd.go:157]: What about wipefs and 'sgdisk -Z'?  Do we have those on RHCOS?

Comment (rollandf) [reply internal/host/installcmd.go:157]: Yes, there are available in RHCOS.
Do you think we need them too?

Comment (avishayt) [reply internal/host/installcmd.go:157]: It's in the ironic code, I think it's worth adding

Comment (rollandf) [reply internal/host/installcmd.go:157]: Sure

**Comment by filanov** [comment]
Can you please make sure that it works on system?

**Comment by rollandf** [comment]
> Can you please make sure that it works on system?

This is why I put it on hold...

**Comment by rollandf** [comment]
Install command:
`
Oct 25 12:19:38 test-infra-cluster-assisted-installer-master-2 agent[1774]: time="25-10-2020 12:19:38" level=info msg="Executing step: <install-d0a85e57>, command: <bash>, args: <[-c wipefs --force --all /dev/vda ; dd if=/dev/zero of=/dev/vda bs=512 count=33 ; dd if=/dev/zero of=/dev/vda bs=512 count=33 seek=$((`blockdev --getsz /dev/vda` - 33)) ; sgdisk -Z /dev/vda ; wipefs --force --all /dev/vdc ; dd if=/dev/zero of=/dev/vdc bs=512 count=33 ; dd if=/dev/zero of=/dev/vdc bs=512 count=33 seek=$((`blockdev --getsz /dev/vdc` - 33)) ; sgdisk -Z /dev/vdc ; podman run -v /dev:/dev:rw -v /opt:/opt:rw -v /run/systemd/journal/socket:/run/systemd/journal/socket --privileged --pid=host --net=host -v /var/log:/var/log:rw --env PULL_SECRET_TOKEN --name assisted-installer quay.io/ocpmetal/assisted-installer:latest --role master --cluster-id b9837bc0-1825-44af-a62f-70b0ff9d9a5a --boot-device /dev/vdb --host-id 1e42fc46-37a5-4d07-8b3a-22299823e80d --openshift-version 4.6 --controller-image quay.io/ocpmetal/assisted-installer-controller:latest --url http://10.1.37.42:6000 --insecure=true --agent-image quay.io/ocpmetal/assisted-installer-agent:latest --host-name test-infra-cluster-assisted-installer-master-2]>" file="step_processor.go:71" request_id=df832b02-2710-486f-a95f-a37d2280b626
`

**Review by filanov — APPROVED** [review]

**Pull request merged**